src/Microsoft.AspNetCore.Server.IISIntegration/Server/IISHttpServer.cs

```diff
@@ -56,15 +62,51 @@ public Task StartAsync<TContext>(IHttpApplication<TContext> application, Cancell
 
         public Task StopAsync(CancellationToken cancellationToken)
         {
-            // TODO: Drain pending requests
+            // First call back into native saying "DON'T SEND ME ANY MORE REQUESTS"
+            NativeMethods.http_stop_incoming_requests();
```

Comment:
Move this down to line 83. The RegisterCancellation code below is designed to prevent StopAsync from executing twice.